### ISSUE TITLE
Segmentation fault on cythonizing

### ISSUE THREAD

**Issue by wkerzendorf** [cython/cython#2199]
We are trying to cythonize our code (github.com/tardis-sn/tardis) within our TRAVIS environment and we get a segmentation fault during the cythonizing (cython 0.27.3 and 0.28) step (see https://travis-ci.org/tardis-sn/tardis/jobs/362581597). 

Here's the log:
```
Compiling tardis/montecarlo/montecarlo.pyx because it changed.
[1/1] Cythonizing tardis/montecarlo/montecarlo.pyx
/home/travis/.travis/job_stages: line 57:  4789 Segmentation fault      (core dumped) CC=$COMPILER python setup.py $SETUP_CMD
```
I'm also pinging @unoebauer as he was helping to debug this problem

**Comment by wkerzendorf** [comment]
@scoder we tried - again segfault: https://travis-ci.org/tardis-sn/tardis/jobs/365640859 - 
it downloads this wheel: 
Downloading https://files.pythonhosted.org/packages/f7/a1/0ecf3ace2e97eaf63f93931be38361763fff0118aca6b70119dde9867e14/Cython-0.28-cp27-cp27mu-manylinux1_x86_64.whl

**Comment by scoder** [comment]
Do I understand correctly that you cannot reproduce this locally, but only on travis?
I tried compiling that .pyx file myself and (unsurprisingly) couldn't get it to crash. And the file looks rather unsuspicious, definitely not something that could run into resource limits or so.

Ok, travis uses an old Ubuntu version, but apart from that, it's entirely non-obvious to me what might trigger a crash here. I've never seen something like this in years. But I admit that I also don't remember seeing anyone use conda on travis yet.

**Comment by wkerzendorf** [comment]
@scoder I've seen many people use conda on TRAVIS - but I have not seen that bug. Not sure where it comes from as we also can not reproduce it locally. I filed a bug on TRAVIS - maybe they can help.

**Comment by wkerzendorf** [comment]
@scoder - I think we are getting close. TRAVIS's @BanzaiMan gave me access to the debug mode and I got a core dump: 

```
Program terminated with signal SIGSEGV, Segmentation fault.
#0  0x00007f17a0579ea4 in _PyObject_GetDictPtr () from /home/travis/miniconda/envs/tardis/bin/../lib/libpython2.7.so.1.0
```

**Comment by wkerzendorf** [comment]
@scoder I've used the cython master in this build and it still crashes: https://travis-ci.org/tardis-sn/tardis/jobs/381891389

**Comment by wkerzendorf** [comment]
`python setup.py build_ext -D CYTHON_CLINE_IN_TRACEBACK=0`

produced this error:

```python
[ 6/13] Cythonizing /home/travis/cython/Cython/Compiler/Pythran.py
[ 7/13] Cythonizing /home/travis/cython/Cython/Compiler/Scanning.py
[ 8/13] Cythonizing /home/travis/cython/Cython/Compiler/Visitor.py
[ 9/13] Cythonizing /home/travis/cython/Cython/Plex/Actions.py
[10/13] Cythonizing /home/travis/cython/Cython/Plex/Scanners.py
[11/13] Cythonizing /home/travis/cython/Cython/Runtime/refnanny.pyx
[12/13] Cythonizing /home/travis/cython/Cython/StringIOTree.py
[13/13] Cythonizing /home/travis/cython/Cython/Tempita/_tempita.py
building 'Cython.Plex.Scanners' extension
creating build
creating build/temp.linux-x86_64-2.7
creating build/temp.linux-x86_64-2.7/home
creating build/temp.linux-x86_64-2.7/home/travis
creating build/temp.linux-x86_64-2.7/home/travis/cython
creating build/temp.linux-x86_64-2.7/home/travis/cython/Cython
creating build/temp.linux-x86_64-2.7/home/travis/cython/Cython/Plex
gcc -pthread -B /home/travis/miniconda/envs/tardis/compiler_compat -Wl,--sysroot=/ -fno-strict-aliasing -g -O2 -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -fPIC -DCYTHON_CLINE_IN_TRACEBACK=0=1 -I/home/travis/miniconda/envs/tardis/include/python2.7 -c /home/travis/cython/Cython/Plex/Scanners.c -o build/temp.linux-x86_64-2.7/home/travis/cython/Cython/Plex/Scanners.o
/home/travis/cython/Cython/Plex/Scanners.c: In function ‘__Pyx_AddTraceback’:
<command-line>:0:28: error: lvalue required as left operand of assignment
/home/travis/cython/Cython/Plex/Scanners.c:1465:53: note: in expansion of macro ‘CYTHON_CLINE_IN_TRACEBACK’
 #define __Pyx_CLineForTraceback(tstate, c_line)  (((CYTHON_CLINE_IN_TRACEBACK)) ? c_line : 0)
                                                     ^
/home/travis/cython/Cython/Plex/Scanners.c:11195:18: note: in expansion of macro ‘__Pyx_CLineForTraceback’
         c_line = __Pyx_CLineForTraceback(tstate, c_line);
                  ^
error: command 'gcc' failed with exit status 1
```

**Comment by scoder** [comment]
I meant something like `CFLAGS="$CFLAGS -D CYTHON_CLINE_IN_TRACEBACK=0"`.

**Comment by wkerzendorf** [comment]
@scoder that worked  - it doesn't crash. What's the best way to get this into the travis version?

**Comment by wkerzendorf** [comment]
@scoder we integrated this into travis in the following way: https://github.com/tardis-sn/tardis/blob/master/.travis.yml (using the compile flags). At what point can we use a standard release version of cython again?

**Comment by scoder** [comment]
Hmm, looks like the fix didn't make it into the changelog. It's included in 0.28.3, though.

**Comment by wkerzendorf** [comment]
@scoder - it still crashes https://travis-ci.org/tardis-sn/tardis/jobs/405443747
